### PR TITLE
Review V1 Swagger (Core Metadata) for completeness.

### LINKED ISSUE #2361: Review V1 Swagger for Completeness -- Core-Metadata
All of the old RAML documents for the API endpoints in Core/Supporting services were converted to Swagger by the QA/Test team. As part of this conversion, we identified the need to review the resulting Swagger for completeness. If the original RAML document did not include comprehensive description of a service's API behavior, then obviously the converted Swagger won't either.

This issue is for review of the core-metadata V1 [Swagger doc](https://github.com/edgexfoundry/edgex-go/blob/master/api/openapi/v1/core-metadata.yaml).

### PR DESCRIPTION
**Pull request by akramtexas**
Fix #2361 
- Please note that this `PR` contains a _large_ number of updates (the Swagger `YAML` consisted of approximately 2500 lines).
- As before (and as with similar updates to other Swagger docs previously), I ran the `YAML` through a validator to ensure that it conforms to being well-formed.

### CODE REVIEW

**Comment by akramtexas** [comment]
Noticed a `Jenkins` build failure related to `Running setup.py install for cbor2` as follows, which may be an intermittent error:
```
[2020-03-19T15:06:20.750Z] ERROR: Command errored out with exit status 1: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-rnp63y9i/cbor2/setup.py'"'"'; __file__='"'"'/tmp/pip-install-rnp63y9i/cbor2/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-wb42eebv/install-record.txt --single-version-externally-managed --compile Check the logs for full command output.
```

**Comment by akramtexas** [comment]
Rebase

**Comment by akramtexas** [comment]
recheck

**Comment by akramtexas** [comment]
Rebased

File: api/openapi/v1/core-metadata.yaml [hunk 1]
```diff
@@ -7,115 +7,102 @@ servers:
...
-        for the Addressable. Returns Internal Service Error (HTTP 500) for unknown
-        or unanticipated issues. Returns NotFoundException (HTTP 404) if no addressable
-        with the provided id is found.
+      description: Remove the addressable designated by the database-generated ID
+        for the addressable.
       parameters:
       - name: id
         in: path
-        description: database generated id
+        description: Database-generated ID
```
Comment (brandonforster) [inline api/openapi/v1/core-metadata.yaml]: no hyphen

File: api/openapi/v1/core-metadata.yaml [hunk 2]
```diff
@@ -171,75 +155,71 @@ paths:
...
-      description: Return Addressable objects with given topic. List may be empty
-        if none are associated to the topic. Returns Internal Service Error (HTTP
-        500) for unknown or unanticipated issues.
+      description: Return addressable objects with given topic. List may be empty
+        if none are associated with the topic.
       parameters:
       - name: topic
         in: path
-        description: topic name (as in MQTT topic name)
+        description: Topic name, for example the MQTT topic name
```
Comment (brandonforster) [inline api/openapi/v1/core-metadata.yaml]: Remove the implementation detail. "Topic name, as used by the messaging provider"

File: api/openapi/v1/core-metadata.yaml [hunk 3]
```diff
@@ -171,75 +155,71 @@ paths:
...
+          description: For malformed or unparsable requests
         500:
           description: for unknown or unanticipated issues.
   /v1/addressable/{id}:
     get:
-      description: Fetch a specific addressable by database generated id. May return
-        null if no addressable matches on id. Returns Internal Service Error (HTTP
-        500) for unknown or unanticipated issues.  Returns NotFoundException (HTTP
-        404) if not found by the id.
+      description: Fetch a specific addressable by database-generated ID. May return
```
Comment (brandonforster) [inline api/openapi/v1/core-metadata.yaml]: "database-generated" is not hyphenated

File: api/openapi/v1/core-metadata.yaml [hunk 4]
```diff
@@ -342,75 +315,65 @@ paths:
...
+          description: If the name is determined to not be unique with regard to others.
         500:
-          description: for unknown or unanticipated issues.
+          description: For unknown or unanticipated issues.
   /v1/device/check/{token}:
     get:
-      description: Obtain a device by either name (preferred) or id. Id must be a
-        valid BSON identifier.
+      description: Obtain a device by either name or ID. ID must be a
+        valid BSON ID.
```
Comment (brandonforster) [inline api/openapi/v1/core-metadata.yaml]: Is this still true? Didn't we switch over to GUIDS?

File: api/openapi/v1/core-metadata.yaml [hunk 5]
```diff
@@ -423,16 +386,13 @@ paths:
...
-          description: device was not found
+          description: Device was not found
   /v1/device/id/{id}:
     delete:
-      description: Remove the Device designated by database generated id. This does
-        not remove associated objects (addressable, service, profile, etc.). Returns
-        Internal Service Error (HTTP 500) for unknown or unanticipated issues. Returns
-        NotFoundException (HTTP 404) if the device cannot be found by the identifier
-        provided.
+      description: Remove the device designated by database-generated ID. This does
```
Comment (brandonforster) [inline api/openapi/v1/core-metadata.yaml]: Another hyphen, please ctrl F the file and find and remove all instances of "database-generated"

File: api/openapi/v1/core-metadata.yaml [hunk 6]
```diff
@@ -443,16 +403,15 @@ paths:
...
+          description: Boolean indicating success of the remove operation
         400:
-          description: for incorrect or unparsable requests
+          description: For incorrect or unparsable requests
         404:
-          description: if the device cannot be found by the id provided.
+          description: If the device cannot be found by the ID provided.
   /v1/device/label/{label}:
     get:
       description: Find all Devices having at least one label matching the label provided.
```
Comment (brandonforster) [inline api/openapi/v1/core-metadata.yaml]: uncapitalize "devices"

File: api/openapi/v1/core-metadata.yaml [hunk 7]
```diff
@@ -604,19 +551,17 @@ paths:
           type: string
       responses:
         200:
-          description: boolean indicating success of update
+          description: Boolean indicating success of update
```
Comment (brandonforster) [inline api/openapi/v1/core-metadata.yaml]: I don't think this is the same phrasing we're using elsewhere for 200 responses.

File: api/openapi/v1/core-metadata.yaml [hunk 8]
```diff
@@ -1976,104 +1805,74 @@ paths:
...
-      description: Return all provision watcher objects sorted by database generated
-        id. Returns Internal Service Error (HTTP 500) for unknown or unanticipated
-        issues. Returns LimitExceededException (HTTP 413) if the number returned exceeds
-        the max limit.
+      description: Return all provision watcher objects sorted by database-generated
+        ID.
       responses:
         200:
-          description: list of provision watcher
+          description: List of provision watcher
```
Comment (brandonforster) [inline api/openapi/v1/core-metadata.yaml]: Make plural

File: api/openapi/v1/core-metadata.yaml [hunk 9]
```diff
@@ -1976,104 +1805,74 @@ paths:
...
-          description: boolean indicating success of the remove operation
+          description: Boolean indicating success of the remove operation
         404:
-          description: if no provision watcher with the provided id is found.
+          description: If no provision watcher with the provided ID is found.
         500:
-          description: for unknown or unanticipated issues
-  /v1/provisionwatcher/identifier/{key}/{value}:
+          description: For unknown or unanticipated issues
+  /v1/provisionwatcher/id/{key}/{value}:
```
Comment (brandonforster) [inline api/openapi/v1/core-metadata.yaml]: This is not correct. There is an "id" AND an "identifier" endpoint.

File: api/openapi/v1/core-metadata.yaml [hunk 10]
```diff
@@ -1976,104 +1805,74 @@ paths:
...
-            with the id or name provided or if the watcher name is determined to not
+          description: If an associated object such as profile, service cannot be found
+            with the ID or name provided or if the watcher name is determined to not
             be unique with regard to others
         500:
-          description: for unknown or unanticipated issues or for any duplicate name
+          description: For unknown or unanticipated issues or for any duplicate name
             (key) error.
   /v1/provisionwatcher/id/{id}:
-    get:
```
Comment (brandonforster) [inline api/openapi/v1/core-metadata.yaml]: Why delete this get section?

File: api/openapi/v1/core-metadata.yaml [hunk 11]
```diff
@@ -1976,104 +1805,74 @@ paths:
...
         200:
-          description: new database generated id for the new ProvisionWatcher
+          description: New database-generated ID for the new provision watcher
         400:
-          description: for malformed or unparsable requests
+          description: For malformed or unparsable requests
         409:
-          description: if an associated object (Profile, Service) cannot be found
-            with the id or name provided or if the watcher name is determined to not
+          description: If an associated object such as profile, service cannot be found
```
Comment (brandonforster) [inline api/openapi/v1/core-metadata.yaml]: "such as the device profile or device service"

File: api/openapi/v1/core-metadata.yaml [hunk 12]
```diff
@@ -2235,18 +2054,41 @@ paths:
...
         400:
           description: For malformed or unparsable requests
         404:
-          description: If no provision watcher is associated with the provided device service, or if the device
-            service or provision watcher cannot be found.
+          description: If no device service match on the ID provided.
         500:
           description: For unknown or unanticipated issues.
+  /v1/provisionwatcher/{id}:
+    get:
```
Comment (brandonforster) [inline api/openapi/v1/core-metadata.yaml]: This is now duplicated

Comment (akramtexas) [reply api/openapi/v1/core-metadata.yaml]: - Cleaned up two relevant end points, and co-located them as well: (1) /v1/provisionwatcher/{id}, which is aligned with a GET, and (2) /v1/provisionwatcher/id/{id}, which is aligned with a DELETE 👍

**Pull request merged**